src/Core/Utilities/CoreHelpers.cs

```diff
@@ -148,6 +148,56 @@ public static DataTable ToArrayTVP(this IEnumerable<SelectionReadOnly> values)
             return table;
         }
 
+        public static DataTable ToTVP<T>(string tableTypeName, IEnumerable<T> values)
+        {
+            var table = new DataTable();
```

Comment:
I just had this hit me... what if we take the performance hit of reflection to build this type, but since compiled types don't change in the same AppDomain at run-time, why not create the base DataTable definition in a static field inside of `CoreHelpers` and re-use that instead of re-building the table definition each time? Not sure if `DataTable` has a `Copy()` method or not, but might be worth it, otherwise no big deal, I'm sure this is quick enough and used infrequently so no harm no foul, just crossed my mind as I was reviewing this again.

Reply:
That's a really interesting idea. We could even go deeper into reflection and reflect out classes that need it because of the `DbOrderAttribute`! (I actually don't think that's worth it for this one class)

Reply:
It might be gnarlier than you were envisioning, but reflection is now front-loaded to a static dictionary in the new `TvpHelpers` static class. 

I was invoking property getters, so the biggest reflection time is in that invocation, not in walking the type definition finding properties. Because of this, I felt I had to include the getters in the front-loading. That means compiled expressions. The end result is a method that's only ~50% slower than directly building the Tvp.

One c# nuance, though, that static isn't calculated until the property is accessed. It will be stored so subsequent calls don't have to go through the reflection.